src/components/with-datasets/index.tsx

```diff
@@ -14,9 +14,11 @@ export interface Props {
 const withDatasets = (Component: ComponentType<any>) => {
   const WrappedComponent = (props: Props) => <Component {...props} />;
 
-  const mapStateToProps = (state: any) => ({
-    datasets: state.DatasetsReducer.get('datasets').toJS()
-  });
+  const mapStateToProps = (state: any, { datasetId }: any) => {
+    return {
+      datasets: state.DatasetsReducer.get(datasetId)?.toJS() ?? []
```

Comment:
Dette ser ganske forvirrende ut. Selve komponent (with-datasets), skal levere en kontrakt, altså eksponere datasets. Når vi forsøker å hente ut datasets basert på datasetId, så gir det feil inntrykk av det komponenten egentlig gjør.

Jeg ser selvfølgelig at komponenten har sine begrensninger, men foreslår likevel at vi ikke benytter oss av caching. Vi tjener ikke noe særlig på det og selve applikasjonen kan bli mindre responsiv i det antall datasett i cachen øker. Enkleste løsning, blir å fyre av en spørring hver gang vi trenger nye datasett.

Reply:
Da må vi rename 'with-datasets' til f.eks. 'with-datasets-for-dataset'. Men mapStateToProps kan jo fint returnere noe annet enn hele state.

mapStateToProps functions can, and should, do a lot more than just return state.someSlice. They have the responsibility of "re-shaping" store data as needed for that component. This may include returning a value as a specific prop name, combining pieces of data from different parts of the state tree, and transforming the store data in different ways. (https://react-redux.js.org/using-react-redux/connect-mapstate).

Så jeg ser ikke at vi bryter pattern med akkurat denne måten å gjøre det på.

Reply:
Vi kan bytte navn, men da er komponenten plutselig minde gjenbrukbar. I dag har den ingen dependencies og eneste jobb den har er å hente hvilke som helst datasett og eksponere de. Eneste jeg ser medmapStateToProps sånn som vi bruker det er semantikk. Når jeg ser kode som henter datasett basert på ID, forventer jeg 1 datasett, mens vi returnerer faktisk flere.

Reply:
Kan endre til å bruke en selector som har et bedre navn når man henter ut på en id, f.eks. datasets: getRelatedDatasetsToDataset(datasetId) ?? [].